src/utils/cspr-network.ts

```diff
@@ -0,0 +1,208 @@
+    ttl: number,
+    contractHash?: string
+  ) {
+    if (this.apiVersion === 2) {
+      new NativeDelegateBuilder()
+        .validator(validatorPublicKey)
+        .from(delegatorPublicKey)
+        .amount(amountMotes)
+        .chainName(networkName)
+        .payment(deployCost)
+        .ttl(ttl)
+        .build();
+    }
+
+    if (contractHash) {
+      return new ContractCallBuilder()
+        .from(delegatorPublicKey)
+        .byHash(contractHash)
+        .entryPoint('delegate')
+        .chainName(networkName)
+        .runtimeArgs(
+          Args.fromMap({
+            validator: CLValue.newCLPublicKey(validatorPublicKey),
+            delegator: CLValue.newCLPublicKey(delegatorPublicKey),
+            amount: CLValueUInt512.newCLUInt512(amountMotes)
+          })
+        )
+        .ttl(ttl)
+        .buildFor1_5();
+    }
+
+    return new Error('Need to provide contract hash');
+  }
+
+  public createUndelegateTransaction(
+    delegatorPublicKey: PublicKey,
+    validatorPublicKey: PublicKey,
+    networkName: string,
+    amountMotes: string | BigNumber,
+    deployCost: number,
+    ttl: number,
+    contractHash?: string
+  ) {
+    if (this.apiVersion === 2) {
+      new NativeUndelegateBuilder()
+        .validator(validatorPublicKey)
+        .from(delegatorPublicKey)
+        .amount(amountMotes)
+        .chainName(networkName)
+        .payment(deployCost)
+        .ttl(ttl)
+        .build();
+    }
+
+    if (contractHash) {
+      return new ContractCallBuilder()
+        .from(delegatorPublicKey)
+        .byHash(contractHash)
+        .entryPoint('undelegate')
+        .chainName(networkName)
+        .ttl(ttl)
+        .runtimeArgs(
+          Args.fromMap({
+            validator: CLValue.newCLPublicKey(validatorPublicKey),
+            delegator: CLValue.newCLPublicKey(delegatorPublicKey),
+            amount: CLValueUInt512.newCLUInt512(amountMotes)
+          })
+        )
+        .buildFor1_5();
+    }
+
+    return new Error('Need to provide contract hash');
+  }
+
+  public createRedelegateTransaction(
+    delegatorPublicKey: PublicKey,
+    validatorPublicKey: PublicKey,
+    networkName: string,
+    amountMotes: string | BigNumber,
+    deployCost: number,
+    ttl: number,
+    contractHash?: string
+  ) {
+    if (this.apiVersion === 2) {
+      new NativeRedelegateBuilder()
+        .validator(validatorPublicKey)
+        .from(delegatorPublicKey)
+        .amount(amountMotes)
+        .chainName(networkName)
+        .payment(deployCost)
+        .ttl(ttl)
+        .build();
+    }
+
+    if (contractHash) {
+      // need to provide contract hash
+      return (
+        new ContractCallBuilder()
+          .from(delegatorPublicKey)
+          .byHash(contractHash)
+          .entryPoint('redelegate')
+          .chainName(networkName)
+          // .amount(amountMotes)
+          .ttl(ttl)
+          .buildFor1_5()
+      );
+    }
+
+    return new Error('Need to provide contract hash');
```

Comment:
Let's change to `Auction contract hash is required when creating a transaction on Casper Network 1.5.x`